xdsl/interactive/app.py

```diff
@@ -213,17 +221,66 @@ def watch_available_pass_list(
                     ListItem(Label(value.name), name=value.name)
                 )
 
+    def get_pass_arguments(self, selected_pass_value: type[ModulePass]) -> None:
+        """
+        This function facilitates user input of pass concatenated_arg_val by navigating
+        to the AddArguments screen, and subsequently parses the returned string upon
+        screen dismissal and appends the pass to the pass_pipeline variable.
+        """
+
+        def add_pass_with_arguments_to_pass_pipeline(concatenated_arg_val: str) -> None:
```

Comment:
can you please make this a method, and pass in a lambda for the callback below?

Reply:
I'm not sure that this is possible. This is a specific thing in textual where when a screen is dismissed, it somehow automatically returns the string we want in the form of the variable (concatenated_arg_val : str) to theadd_pass_with_arguments_to_pass_pipeline function that is declared in the push_screen method self.push_screen(screen, add_pass_with_arguments_to_pass_pipeline). It's all connected and cannot be made as a method I think.

See 'Returning Data from Screens' Section here:
https://textual.textualize.io/guide/screens/#returning-data-from-screens

Reply:
Sure that makes sense, it just feels like a part of the logic might be better in a method. Re-reading it some more, I'm not sure exactly how I would do this myself, so maybe it's better to merge as-is and see how it evolves later.

Reply:
Sounds great - we can discuss further in future.